Stored computed columns support FKs, with limits (#18792)
cockroachdb/docs: src/current/v24.2/computed-columns.md

- Cannot be used to generate other computed columns.
- Behave like any other column, with the exception that they cannot be written to directly.
- Are mutually exclusive with [`DEFAULT`]({% link {{ page.version.version }}/default-value.md %}) and [`ON UPDATE`]({% link {{ page.version.version }}/create-table.md %}#on-update-expressions) expressions.

> - Can be used in [`FOREIGN KEY`]({% link {{ page.version.version }}/foreign-key.md %}) constraints, but restricted to the following subset of supported options. The restriction to this subset is necessary because we cannot allow the computed column value to change. Prior to this change, stored computed columns could not be used in foreign key constraints at all.

Reviewer: Should there be a "new in 24.2" tag here? I think the last sentence should be cut because it reads like a release note and will likely confuse in subsequent versions.

```suggestion
- Can be used in [`FOREIGN KEY`]({% link {{ page.version.version }}/foreign-key.md %}) constraints, but are restricted to the following subset of supported options. This restriction is necessary because we cannot allow the computed column value to change.
```

Author: good points both

updated the text as you suggested

also added the "new in" thingy 

PTAL!
